src/interp/interp.h

```diff
@@ -523,14 +522,35 @@ union Value {
   template <typename T>
   T WABT_VECTORCALL Get() const;
   template <typename T>
+  T& WABT_VECTORCALL GetRef();
+  template <typename T>
   void WABT_VECTORCALL Set(T);
 
-  u32 i32_;
-  u64 i64_;
-  f32 f32_;
-  f64 f64_;
-  v128 v128_;
-  Ref ref_;
+ private:
+  union {
+    u32 i32_;
+    u64 i64_;
+    f32 f32_;
+    f64 f64_;
+    v128 v128_;
+    Ref ref_;
+  };
+
+ public:
+#ifndef NDEBUG
+  Value() : v128_(0, 0, 0, 0), type(ValueType::Any) {}
+  void SetType(ValueType t) { type = t; }
+  void CheckType(ValueType t) const {
+    // Sadly we must allow Any here, since locals may be uninitialized.
+    // Alternatively we could modify InterpAlloca to set the type.
+    assert(t == type || type == ValueType::Any);
```

Comment:
This is going to start to get trickier with newer proposals (comparing references, etc.), but for now it's a good check.